q2_feature_table/tests/filter/test_conditional_filter.py

```diff
@@ -40,6 +40,48 @@ def test_conditional(self):
         npt.assert_array_equal(known.ids(axis='observation'),
                                test_.ids(axis='observation'))
 
+    def test_allow_empty_table_true(self):
+        # test True
+        table = biom.Table(
+            data=np.array([[0,   0,  10,   0,   0],
+                           [250, 250, 140,  90, 150],
+                           [250,  25, 100, 200, 100],
+                           [0, 225, 250, 210, 250]]),
+            sample_ids=['A', 'B', 'C', 'D', 'E'],
+            observation_ids=['bat', 'cat', 'rat', 'a-tat-tat']
+            )
+        known = biom.Table(
+            data=np.array([[0, 225, 250, 210, 250]]),
+            sample_ids=['A', 'B', 'C', 'D', 'E'],
+            observation_ids=['a-tat-tat']
+            )
+        test_ = filter_features_conditionally(table,
+                                              prevalence=0.8,
+                                              abundance=0.4,
+                                              allow_empty_table=True)
+        npt.assert_array_equal(known.matrix_data.toarray(),
+                               test_.matrix_data.toarray())
+        npt.assert_array_equal(known.ids(axis='sample'),
+                               test_.ids(axis='sample'))
+        npt.assert_array_equal(known.ids(axis='observation'),
+                               test_.ids(axis='observation'))
+
+    def test_allow_empty_table_false(self):
+        # test False
+        table = biom.Table(
+            data=np.array([[0,   0,  10,   0,   0],
+                           [250, 250, 140,  90, 150],
+                           [250,  25, 100, 200, 100],
+                           [0, 225, 250, 210, 250]]),
+            sample_ids=['A', 'B', 'C', 'D', 'E'],
+            observation_ids=['bat', 'cat', 'rat', 'a-tat-tat']
+            )
+        with self.assertRaises(ValueError):
```

Comment:
When adding these kinds of assertions, it's best to be as specific as possible - what I'd recommend doing instead is using an `assertRaisesRegex` that contains a segment of that specific ValueError's text. That way we know we're hitting the expected error branch!

Reply:
That makes sense! I will do that. Thank you!